### PR TITLE
feat(docs): Add OTel eBPF (OBI) doc

### LINKED ISSUE #2832: Support ebpf
## Is your feature request related to a problem?

Given docker containers that contain a c++ game server executable, I don't want to instrument to get observability in SigNoz.

Here's a detailed problem:

In Godot Engine, we have a VOIP server in a docker container. There are clicks and pops in the opus compression in the voip layer.

We want to get an idea of occurrences of these failures but to solve the quality issues require observability.

For SigNoz to help, the solution requires a open telemetry c++ sdk and random places in the docker container which is a lot of effort to evaluate SigNoz.

If via eBPF, we can runtime insert probes and observe without recompiling, that would make the debugging easier.

To gain observability, it may not be possible to log via instrumentation efficiently enough and would requires a kernel module like ebpf.

## Describe the solution you'd like

Support ebpf in SigNoz for observability.

## Describe alternatives you've considered

1. https://github.com/cilium/hubble
2. https://github.com/Flowmill/flowmill-collector
3. https://github.com/pixie-labs/pixie
4. https://github.com/cloudflare/ebpf_exporter
5. https://www.percona.com/blog/2020/07/06/adding-ebpf-based-metrics-to-percona-monitoring-and-management/
6. https://docs.projectcalico.org/about/about-ebpf

## Additional context

I believe it's possible to make an ebpf collector demo via https://github.com/Flowmill/flowmill-collector and the OpenTelemetry Collector.

### PR DESCRIPTION
**Pull request by crazyuploader**
## Pull Request

---

### 📄 Summary
This PR adds support for OTel eBPF (OBI) doc.

Doc: [Link](https://staging.signoz.io/docs/instrumentation/opentelemetry-ebpf)

#### Issues closed by this PR
Closes: https://github.com/SigNoz/signoz.io/issues/2832
---

### ✅ Change Type
_Select all that apply_

- [ ] ✨ Feature
- [ ] 🐛 Bug fix
- [ ] ♻️ Refactor
- [ ] 🛠️ Infra / Tooling
- [ ] 🧪 Test-only

---

---

- Tests added/updated:
- Manual verification:
- Edge cases covered:

---

- Blast radius:
- Potential regressions:
- Rollback plan:

---

| Field | Value |
|------|-------|
| Deployment Type | Cloud / OSS / Enterprise |
| Change Type | Feature / Bug Fix / Maintenance |
| Description | User-facing summary |

---

### 📋 Checklist
- [ ] Tests added or explicitly not required
- [ ] Manually tested
- [ ] Breaking changes documented
- [ ] Backward compatibility considered

---

## 👀 Notes for Reviewers

---

### CODE REVIEW

File: data/docs/instrumentation/opentelemetry-ebpf.mdx [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+
+### Step 1. Run OBI against your service
+
+Choose your environment.
+
+<Tabs entityName="environment">
+
+<TabItem value="docker" label="Docker" default>
+
+Run OBI as a sidecar container alongside the container you want to observe. OBI needs the host PID namespace and elevated privileges to load eBPF probes.
```
Comment (revmag) [inline data/docs/instrumentation/opentelemetry-ebpf.mdx]: Can we link some relevant docs here?

File: data/docs/instrumentation/opentelemetry-ebpf.mdx [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+      OTEL_EXPORTER_OTLP_ENDPOINT: "https://ingest.<region>.signoz.cloud:443"
+      OTEL_EXPORTER_OTLP_HEADERS: "signoz-ingestion-key=<your-ingestion-key>"
+      OTEL_EXPORTER_OTLP_PROTOCOL: "http/protobuf"
+```
+
+</TabItem>
+
+<TabItem value="kubernetes" label="Kubernetes">
+
+Deploy OBI as a DaemonSet so one instance runs on each node. Use this manifest to set host PID access, grant RBAC read access for pod metadata, and read the ingestion key from a Secret.
```
Comment (revmag) [inline data/docs/instrumentation/opentelemetry-ebpf.mdx]: Link some relevant docs

File: data/docs/instrumentation/opentelemetry-ebpf.mdx [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+</Admonition>
+
+## Prerequisites
+
+- Linux kernel **5.8+** with BTF enabled (RHEL 4.18+ works too).
+- Root or `--privileged`, or the [required Linux capabilities](#run-obi-with-reduced-privileges).
+- A service that serves HTTP, HTTPS, or gRPC on a known port.
+- An instance of SigNoz (either [Cloud](https://signoz.io/teams/) or [Self-Hosted](https://signoz.io/docs/install/self-host/)).
+
+## Select the target process
```
Comment (revmag) [inline data/docs/instrumentation/opentelemetry-ebpf.mdx:31]: This is not very clear

File: data/docs/instrumentation/opentelemetry-ebpf.mdx [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+
+Tell OBI which process to instrument with one of these selectors, then reuse it in the steps below:
+
+- **By port** with `OTEL_EBPF_OPEN_PORT`: the process listening on a port or range, such as **8443** or **8000-8999**.
+- **By executable** with `OTEL_EBPF_AUTO_TARGET_EXE`: an executable path glob, such as **\*/myserver**.
+
+The Docker and Linux host examples use `OTEL_EBPF_OPEN_PORT`. The Kubernetes DaemonSet example uses `OTEL_EBPF_AUTO_TARGET_EXE` for node-wide process discovery.
```
Comment (revmag) [inline data/docs/instrumentation/opentelemetry-ebpf.mdx]: Can Docker and LInux use  OTEL_EBPF_AUTO_TARGET_EXE ? and vice versa?

**Review by revmag — APPROVED** [review]
